script.js

```diff
@@ -1,12 +1,304 @@
+  rootElem.appendChild(theContainer); // Appended the div element to the parent element of root.
+
+  let theArticle = document.createElement("article"); // Creates an article element.
+  theArticle.id = "article"; //Assigned the id of article to the article.
+  theArticle.setAttribute("class", "artContainer"); // Assigned the class of artContainer to the Article element.
+  theContainer.appendChild(theArticle); // Appended the Article to the div of our placeholder.
+
+  let theTitleTagName = document.createElement("h3"); //Created the a h3 element to hold the title.
+  theArticle.appendChild(theTitleTagName); // Appended the h3 element to the article Container.
+  theTitleTagName.textContent = `${element.name}-${titleMaker(
+    element.season,
+    element.number
+  )}`; // Set the the text content of the h3 element the object name property called the title maker maker function
+  theTitleTagName.id = "title"; // Set the id of h3 element to 'title'.
+
+  let theImageTagName = document.createElement("img"); // created an image element.
+  theArticle.appendChild(theImageTagName); // Appended the image tag to the article parents.
+  theImageTagName.id = "image"; //Set an id of 'image to it'
+
+  var noImage = "";
+  element.image == null ? noImage : (noImage = element.image.medium);
+  theImageTagName.src = noImage;
+  //Assigned the medium image object property to img element.
+  let theParagraph = document.createElement("p"); // Created Paragraph element.
+
+  theParagraph.textContent = element.summary.replace(/(<([^>]+)>)/gi, ""); //set the paragraph text content to the summary object property and remove the preset p tag using replace method
+  theParagraph.id = "summary"; //Set the id of summary to the paragraph element.
+  theArticle.appendChild(theParagraph); // Appended the paragraph to the article
+}
+
+/*5 fixes the title*/
+/* This function will display the season number correctly takes season and number as a parameter*/
+/**/
+function titleMaker(season, number) {
+  return `S0${season}E${number < 10 ? "0" + number : number}`; // interpolates the season 'S0' and checks if number is less than 10, concat '0' to number else returns a number.
+}
+
+/*This function will search the input and takes input as a parameter */
+
+/*6 creates search*/
+function search(input) {
+  let searchResult = allEpisodes.filter((episode) => {
+    // This function will filter all episode array of objects take one episode object as an element.
+    return (
+      episode.name.toLowerCase().includes(input.toLowerCase()) || //Compares object property of name that is lowercase is === to the input that the user enters if not converts it to the lowercase.
+      episode.summary.toUpperCase().includes(input.toUpperCase()) //Compares object property of summary that is Uppercase is === to the input that the user enters if not converts it to the Uppercase.
```

Comment:
nice solution, bravo.

Just for fun, here is another way to do this - something to think about

```js
let stringToSearch = `${episode.name} ${episode.summary}`.toLowerCase()
return stringToSearch.includes(input.toLowerCase())
```

Can you see how this one works the same way?